hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -192,6 +273,63 @@ class IRSuite extends HailSuite {
     assertEvalsTo(ApplyUnaryPrimOp(BitNot(), i64na), null)
   }
 
+  @Test def testApplyUnaryPrimOpInferPType() {
+    val i32na = NA(TInt32())
+    val i64na = NA(TInt64())
+    val f32na = NA(TFloat32())
+    val f64na = NA(TFloat64())
+    val bna = NA(TBoolean())
+
+    var node = ApplyUnaryPrimOp(Negate(), I32(5))
+    assertPType(node, PInt32())
+    node = ApplyUnaryPrimOp(Negate(), i32na)
+    // TODO: this feels like there should be a NULL/missing type, or at least a missing flag on the PType
+    assertPType(node, PInt32())
+
+    // i32na is already inferred which breaks inference of the new ApplyUnaryPrimOp node
```

Comment:
make `i32na` and friends defs to avoid this issue

Reply:
I wrote this explicitly because I didn't find the behavior explicitly stated if I made that a def, or say NA(i32()). 

I know comments here are not allowed. At the same time we don't have centralized documentation on our compiler. Mind if I start, and you contribute?

Reply:
comments should be fine if they indicate something not obvious from the code! I just mean you can fix this problem by making the things defs

Reply:
maybe with a comment about why they're defs :)

Reply:
I'm more than slightly worried about my next 1:1 with @cseed after I add a comment to the codebase :)

K will do!